benchmark/rocblas/blas2/gbmv.cpp

```diff
@@ -0,0 +1,216 @@
+  state.counters["ku"] = ku_d;
+
+  // Compute the number of A non-zero elements.
+  const double A_validVal =
+      (t_str[0] == 'n' ? n_d : m_d) * (kl_d + ku_d + 1.0) -
+      0.5 * (kl_d * (kl_d + 1.0)) - 0.5 * (ku_d * (ku_d + 1.0));
+
+  {
+    double nflops_AtimesX = 2.0 * A_validVal;
+    double nflops_timesAlpha = ylen;
+    double nflops_addBetaY = (beta != scalar_t{0}) ? 2 * ylen : 0;
+    state.counters["n_fl_ops"] =
+        nflops_AtimesX + nflops_timesAlpha + nflops_addBetaY;
+  }
+  {
+    double mem_readA = A_validVal;
+    double mem_readX = xlen;
+    double mem_writeY = ylen;
+    double mem_readY = (beta != scalar_t{0}) ? ylen : 0;
+    state.counters["bytes_processed"] =
+        (mem_readA + mem_readX + mem_writeY + mem_readY) * sizeof(scalar_t);
+  }
+
+  // Matrix options (rocBLAS)
+  const rocblas_operation transA =
+      t_str[0] == 'n' ? rocblas_operation_none : rocblas_operation_transpose;
+
+  // Data sizes
+  const int m_size = lda * n;
+  const int v_x_size = 1 + (xlen - 1) * incX;
+  const int v_y_size = 1 + (ylen - 1) * incY;
+
+  // Input matrix/vector, output vector.
+  std::vector<scalar_t> m_a =
+      blas_benchmark::utils::random_data<scalar_t>(m_size);
+  std::vector<scalar_t> v_x =
+      blas_benchmark::utils::random_data<scalar_t>(v_x_size);
+  std::vector<scalar_t> v_y =
+      blas_benchmark::utils::random_data<scalar_t>(v_y_size);
+
+  {
+    // Device memory allocation & H2D copy
+    blas_benchmark::utils::HIPVector<scalar_t> m_a_gpu(m_size, m_a.data());
+    blas_benchmark::utils::HIPVector<scalar_t> v_x_gpu(v_x_size, v_x.data());
+    blas_benchmark::utils::HIPVector<scalar_t> v_y_gpu(v_y_size, v_y.data());
+
+    CHECK_ROCBLAS_STATUS(
+        rocblas_set_pointer_mode(rb_handle, rocblas_pointer_mode_host));
+
+#ifdef BLAS_VERIFY_BENCHMARK
+    // Reference gbmv
+    std::vector<scalar_t> v_y_ref = v_y;
+    reference_blas::gbmv(t_str, m, n, kl, ku, alpha, m_a.data(), lda,
+                         v_x.data(), incX, beta, v_y_ref.data(), incY);
+
+    // Rocblas verification gbmv
+    std::vector<scalar_t> v_y_temp = v_y;
+    {
+      // Temp result on device (copied back to Host upon destruction)
+      blas_benchmark::utils::HIPVector<scalar_t, true> y_temp_gpu(
+          ylen, v_y_temp.data());
+      // rocBLAS function call
+      rocblas_gbmv_f<scalar_t>(rb_handle, transA, m, n, kl, ku, &alpha, m_a_gpu,
+                               lda, v_x_gpu, incX, &beta, y_temp_gpu, incY);
+    }
+
+    std::ostringstream err_stream;
+    if (!utils::compare_vectors(v_y_temp, v_y_ref, err_stream, "")) {
+      const std::string& err_str = err_stream.str();
+      state.SkipWithError(err_str.c_str());
+      *success = false;
+    };
+#endif
+
+    auto blas_warmup = [&]() -> void {
+      rocblas_gbmv_f<scalar_t>(rb_handle, transA, m, n, kl, ku, &alpha, m_a_gpu,
+                               lda, v_x_gpu, incX, &beta, v_y_gpu, incY);
+      CHECK_HIP_ERROR(hipStreamSynchronize(NULL));
```

Comment:
Could you move this `hipStreamSynchronize` to line 168 as in level1 operators?

Reply:
Yes missed that one! Thanks.